### PR TITLE
add teardown when TestExecPod finished

### PR DESCRIPTION
**Pull request by googs1025**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
- add teardown when TestExecPod finished
#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
None
```

#### Additional documentation e.g., Usage docs, etc.:

```docs
None
```

### CODE REVIEW

File: examples/pod_exec/envtest_test.go [hunk 1]
```diff
@@ -78,6 +78,13 @@ func TestExecPod(t *testing.T) {
 				t.Fatal("Couldn't connect to en.wikipedia.org")
 			}
 			return ctx
+		}).
+		Teardown(func(ctx context.Context, t *testing.T, c *envconf.Config) context.Context {
+			dep := ctx.Value("test-deployment").(*appsv1.Deployment)
```
Comment (googs1025) [inline examples/pod_exec/envtest_test.go:83]: here ci failed because it can't get the information

**Comment by vladimirvivien** [comment]
@googs1025 Thanks for the work. Is this meant to be a fix in the framework code or an update in the example only ? I only see an update to the example code.

**Comment by googs1025** [comment]
> @googs1025 Thanks for the work. Is this meant to be a fix in the framework code or an update in the example only ? I only see an update to the example code.

Yes, I just modified the code in the example.

**Comment by googs1025** [comment]
> @googs1025 ok thanks. Since this is just an example. Can you retitle the PR and remove "fix" ? I thought this was a fix for a bug. Thanks.

OK, thanks for the comment. done

**Pull request merged**